software/image_processor.py

```diff
@@ -106,7 +112,19 @@ def analyze_balls(self, t_balls, depth, fragments) -> list:
             if depth is None:
                 obj_dst = obj_y
             else:
-                obj_dst = np.average(depth[obj_y-2:obj_y+2, obj_x-2:obj_x+2])
+                try:
+                    obj_dst = np_average_jit(
+                        depth[obj_y-2:obj_y+2, obj_x-2:obj_x+2])
+                except (ZeroDivisionError):
+                    self.logger.log.error(
+                        "Ball attempted to divide by zero when averaging.")
+                    continue
+
+            # don't add if ball is further than the basket or too close to it
+            if basket != None:
```

Comment:
What units are these. Maybe they could be related to the camera resolution instead of being constants

Reply:
The line comparing basket.x is pixels, and the second one is depth camera distance. 

I might just rewrite or remove this entire thing though, because currently we have issues with the robot getting stuck in front of a basket due to finding the ball for a frame and then losing it again.

Reply:
Unhardcoded them.